docs/_docs/reference/contextual/context-bounds.md

```diff
@@ -4,50 +4,208 @@ title: "Context Bounds"
 
-If the source version is `future-migration`, any pairing of an evidence
+ 1. If one of the bounds is referred to by its name in a subsequent parameter clause, the context bounds are mapped to a using clause immediately preceding the first such parameter clause.
+ 2. Otherwise, if the last parameter clause is a using (or implicit) clause, merge all parameters arising from context bounds in front of that clause, creating a single using clause.
+ 3. Otherwise, the parameters arising from context bounds form a new using clause at the end.
+
+Rules (2) and (3) match Scala 2's rules. Rule (1) is new but since context bounds so far could not be referred to, it does not apply to legacy code. Therefore, binary compatibility with Scala 2 and earlier Scala 3 versions is maintained.
+
+**Examples:**
+
+ 1. By rule 3,
+    ```scala
+    def f[T: {C1, C2}](x: T): R
+    ```
+    expands to
+    ```scala
+    def f[T](x: T)(using C1, C2): R
+    ```
+    Equally by rule 3,
+    ```scala
+    def f[T: {C1 as c1, C2 as c2}](x: T): R
+    ```
+    expands to
+    ```scala
+    def f[T](x: T)(using c1: C1, c2: C2): R
+
+ 2. By rule 2,
+    ```scala
+    def f[T: {C1, C2}, U: C3](x: T)(using y: U, z: V): R
+    ```
+    expands to
+    ```scala
+    def f[T, U](x: T)(using _: C1[T], _: C2[T], _: C3[U], y: U, z: V): R
+    ```
+    The same expansion occurs if `y` and `z` are Scala 2 style `implicit` parameters.
+ 3. Assume the following trait definition:
+    ```scala
+    trait Parser[P]:
+      type Input
+      type Result
+    ```
+    Here is a method `run` that runs a parser on an input of the required type:
+    ```scala
+    def run[P : Parser as p](in: p.Input): p.Result
+    ```
+    By rule 1, this method definition is expanded to:
+    ```scala
+    def run[P](using p: Parser[P]](in: p.Input): p.Result
+    ```
+    Note that the `using` clause is placed in front of the explicit parameter clause `(in: p.Result)` so that
+    the type `p.Result` can legally refer to the context parameter `p`.
+
+### Migration
+
+To ease migration, context bounds map in Scala 3.0 - 3.5 to old-style implicit parameters
+for which arguments can be passed either with a `(using ...)` clause or with a normal application. From Scala 3.6 on, they will map to context parameters instead, as is described above.
+
+If the source version is `3.6-migration`, any pairing of an evidence
 context parameter stemming from a context bound with a normal argument will give a migration
 warning. The warning indicates that a `(using ...)` clause is needed instead. The rewrite can be
 done automatically under `-rewrite`.
 
+## Context Bounds for Polymorphic Functions
+
+From Scala 3.6 on, context bounds can also be used in polymorphic function types and polymorphic function literals:
+
+```scala
+type Comparer = [X: Ord] => (x: X, y: X) => Boolean
+val less: Comparer = [X: Ord as ord] => (x: X, y: X) =>
+  ord.compare(x, y) < 0
+```
```

Comment:
I think this example is either not correct - it does not compile in neither plain Scala 3.6-nightly or with experimental:modularity. 
It fails to parse by the compiler

Reply:
yes that one is still in the works: https://github.com/scala/scala3/pull/21643